ipa-core/src/report/hybrid.rs

```diff
@@ -341,4 +606,52 @@ mod test {
         let expected_err = unique_bytes.check_duplicates(&[bytes4, bytes3]);
         assert!(matches!(expected_err, Err(Error::DuplicateBytes(6))));
     }
+
+    #[test]
+    fn serialization_hybrid_impression() {
```

Comment:
Can a test be added that tests serialization against a pre-existing value? 

This test is making sure that serialize and deserialize are mirrored but if both methods make the same mistake then you wouldn't know.

Reply:
Hmm, good point. I see that ipa.rs has some tests that use constants. Do you happen to know how those were generated?

Reply:
I'm not sure, but I would venture to guess they we're generated by the current implementation. @cberkhoff do you have any ideas for how to accomplish what you're suggesting?

Reply:
Can't we do something simple like?

```
let input = "<hybrid-report>";
assert_eq!(serialize(input), expectation);
```

Reply:
If expectation is generated using the serialize function, wouldn't you have the same potential issue of both methods making the same mistake?

Reply:
My idea was that the input string would be "set in stone" and easy to see and review. Since you would be writing it down you can make sure it's exactly what you want it to be. I imagine you would use serialize to do this once and then copy paste it.

Reply:
I just added a few lines to the end of the test to show deserialzation from a bytestring. Let me know if this is what you had in mind

Reply:
I think this is ok. Maybe testing serialization would have been more legible (my bad, I did mention deserialization before).

Also having this as a separate/independent test would make it easier to triage if issues happen.